### PR TITLE
feat: [CDS-29163]: JIRA create/update connector invalid handled

### PR DESCRIPTION
**Pull request by naman-goenka**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/36100)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 1 Issues 
* Code issue 1 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/36100)

File: pipeline-service/service/src/main/java/io/harness/pms/jira/JiraStepHelperServiceImpl.java [hunk 1]
```diff
@@ -68,6 +69,13 @@ public TaskRequest prepareTaskRequest(JiraTaskNGParametersBuilder paramsBuilder,
     NGAccess ngAccess = AmbianceUtils.getNgAccess(ambiance);
     IdentifierRef identifierRef = IdentifierRefHelper.getIdentifierRef(
         connectorRef, ngAccess.getAccountIdentifier(), ngAccess.getOrgIdentifier(), ngAccess.getProjectIdentifier());
+    if (NULL_VALUE.equals(identifierRef.getIdentifier())) {
```
Comment (puthrayaharness) [inline pipeline-service/service/src/main/java/io/harness/pms/jira/JiraStepHelperServiceImpl.java:72]: We cannot do this. We currently do not restrict customers from creating project level connectors wtih `null` as identifiers.

**Review by puthrayaharness — CHANGES_REQUESTED** [review]

File: pipeline-service/service/src/main/java/io/harness/pms/jira/JiraStepHelperServiceImpl.java (hunk 1 above)
Comment (naman-goenka) [reply pipeline-service/service/src/main/java/io/harness/pms/jira/JiraStepHelperServiceImpl.java:72]: We cannot allow this as then we would be making invalid call to connector client, we have to gracefully fail in this case. Same check is used here as well `pipeline-service/service/src/main/java/io/harness/pms/servicenow/ServiceNowStepHelperServiceImpl.java`

**Review by puthrayaharness — DISMISSED** [review]
> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36100)* status: :shipit: complete! ✓

**Review by archit-harness — DISMISSED** [review]

**Review by puthrayaharness — APPROVED** [review]
> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36100)* status: :shipit: complete! ✓

**Review by archit-harness — APPROVED** [review]
> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36100)* status: :shipit: complete! ✓

**Pull request merged**